k8s-operator/sessionrecording/hijacker.go

```diff
@@ -137,8 +138,7 @@ func (h *Hijacker) setUpRecording(ctx context.Context, conn net.Conn) (net.Conn,
 		errChan <-chan error
 	)
 	h.log.Infof("kubectl exec session will be recorded, recorders: %v, fail open policy: %t", h.addrs, h.failOpen)
-	// TODO (irbekrm): send client a message that session will be recorded.
-	wc, _, errChan, err = h.connectToRecorder(ctx, h.addrs, h.ts.Dial)
+	wc, _, errChan, err = h.connectToRecorder(ctx, h.addrs, h.ts.Dial, h.log.Infof)
```

Comment:
I'm not totally sure I prefer this, but an alternative implementation to consider, we could use `httptrace` here, and remove the need to pass the logger all the way down:

```suggestion
var recorderAddr net.Addr
	trace := &httptrace.ClientTrace{
		GotConn: func(info httptrace.GotConnInfo) {
			recorderAddr = info.Conn.RemoteAddr()
		},
	}

	wc, _, errChan, err = h.connectToRecorder(httptrace.WithClientTrace(ctx, trace), h.addrs, h.ts.Dial)
```

This might make it easier to control exactly when we want to log that info and how much extra metadata we put into that log line?

Reply:
@tomhjp I tested this out and it's great,as you said don't need to pass logger around + can easily log both recorder address as well as target details.
I've tested this with a list of recorder addresses where some are invalid to ensure that it picks up the right one- works as expected:

```
{"level":"info","ts":"2024-09-06T17:36:04Z","logger":"apiserver-proxy","msg":"exec session to container \"nettools\" in Pod \"proxy\" namespace \"default\" will be recorded, the recording will be sent to a tsrecorder instance at \"[fd7a:115c:a1e0::6b01:8b4a]:80\""}
```
I've updated this Pr with the change